### PR TITLE
fix: solidity coding style

### LINKED ISSUE #835: fix: Solidity Coding Style Guideline Violations
# Description

- Following the Solidity Style Guide wherever possible is advisable, especially when it comes to variable name conventions and code layout.
- This can be consistently enforced across the entire code base by means of an automatic linter, such as Solhint.

# Examples

- Variable names should begin lower case. Type declarations (including Contracts) begin upper case.
- Declare modifiers before functions and don’t mix them.

### LINKED ISSUE #837: fix: Unused Imports
# Description

- Conduct a thorough review of all contracts to identify and eliminate any other unused imports or missing interface implementations.

# Examples

- `AlignedLayerServiceManager` unused `IPauserRegistry`, unused `Pausable`.

### PR DESCRIPTION
**Pull request by NicolasRampoldi**
# Description

- Adds `solhint` linter to the CI. To run `solhint` locally, go to `contracts`, run `npm install` and then `npm run lint:sol`. The CI will run the linter but won't stop the user from pushing to main, it just lists the warnings.
- Fixes most linter warnings, the ones left don't seem relevant.
- The three rules added were taken from [https://github.com/lambdaclass/era-contracts/blob/main/.solhint.json](https://github.com/lambdaclass/era-contracts/blob/main/.solhint.json) and it made sense to add them to this repo too.
- Add `make lint_contracts` to makefile

# To Test

- Try deploying and test normal flow.
- Make some errors on purpose to see if the refactoring of requires to custom errors works properly.

### CODE REVIEW

**Review by uri-99 — COMMENTED** [review]
.

**Review by uri-99 — CHANGES_REQUESTED** [review]
Everything seems to be working fine, and read it over for quite some time, but we should talk about this PR because it is pretty big and potentially dangerous.

I used to like a lot the `require` statements, but custom errors are pretty powerful, mainly because of accepting parameters. More info on this: https://soliditylang.org/blog/2021/04/21/custom-errors/ 

It looks like it should be compatible with an upgrade, with no need of changing the GAP, but this should be tested more in detail just in case.

Also, should errors be declared in the Interface? While we're at it, Events could also be declared in the Interface.

Also I believe this PR should require at least 3 or even 4 approves.

Finally, a `make` target to run solhint should come in pretty handy.

File: contracts/src/core/BatcherPaymentService.sol [hunk 1]
```diff
@@ -220,9 +261,13 @@ contract BatcherPaymentService is
...
         if (leaves.length == 1) {
-            require(leaves[0] == batchMerkleRoot, "Invalid merkle root");
+            if (leaves[0] != batchMerkleRoot) {
+                revert InvalidMerkleRoot(batchMerkleRoot, leaves[0]);
+            }
         } else {
-            require(layer[0] == batchMerkleRoot, "Invalid merkle root");
+            if (layer[0] != batchMerkleRoot) {
+                revert InvalidMerkleRoot(batchMerkleRoot, layer[0]);
+            }
```
Comment (taturosati) [inline contracts/src/core/BatcherPaymentService.sol]: Change to else if instead of nested

File: contracts/src/core/BatcherPaymentService.sol [hunk 2]
```diff
@@ -109,38 +155,42 @@ contract BatcherPaymentService is
...
+        if (userData[msg.sender].balance <= 0) {
+            revert UserHasNoFundsToLock(msg.sender);
+        }
+        userData[msg.sender].unlockBlock = 0;
     }
 
     function withdraw(uint256 amount) external whenNotPaused {
-        UserInfo storage user_data = UserData[msg.sender];
-        require(user_data.balance >= amount, "Payer has insufficient balance");
+        UserInfo storage user_data = userData[msg.sender];
```
Comment (taturosati) [inline contracts/src/core/BatcherPaymentService.sol]: user_data should probably be senderData

File: contracts/src/core/BatcherPaymentService.sol [hunk 3]
```diff
@@ -236,30 +281,38 @@ contract BatcherPaymentService is
...
 
         address signer = noncedHash.recover(signatureData.signature);
-        require(signer != address(0), "Invalid signature");
 
-        UserInfo storage user_data = UserData[signer];
+        if (signer == address(0)) {
+            revert InvalidSignature();
+        }
+
+        UserInfo storage user_data = userData[signer];
```
Comment (taturosati) [inline contracts/src/core/BatcherPaymentService.sol]: Same here, should probably be signerData or signerInfo or similar

**Review by taturosati — CHANGES_REQUESTED** [review]

File: contracts/src/core/BatcherPaymentService.sol (hunk 1 above)
Comment (NicolasRampoldi) [reply contracts/src/core/BatcherPaymentService.sol]: Solved in [c9b32e9](https://github.com/yetanotherco/aligned_layer/pull/843/commits/c9b32e95f5fd91f19c54dfb4ed6d97655e7da8d5)

File: contracts/src/core/BatcherPaymentService.sol (hunk 2 above)
Comment (NicolasRampoldi) [reply contracts/src/core/BatcherPaymentService.sol]: Solved in [a0fdcd2](https://github.com/yetanotherco/aligned_layer/pull/843/commits/a0fdcd20075b73bc1ac1cdc0a66ae28d1eb89b2a)

File: contracts/src/core/BatcherPaymentService.sol (hunk 3 above)
Comment (NicolasRampoldi) [reply contracts/src/core/BatcherPaymentService.sol]: Solved in [0191794](https://github.com/yetanotherco/aligned_layer/pull/843/commits/01917949f261623204b62054612eeace2527c0cf).

**Review by taturosati — APPROVED** [review]
Code lgtm, worked in my machine

**Review by uri-99 — APPROVED** [review]

**Pull request merged**